src/Akeneo/Tool/Component/Classification/Updater/CategoryUpdater.php

```diff
@@ -23,19 +25,20 @@
  */
 class CategoryUpdater implements ObjectUpdaterInterface
 {
-    /** @var PropertyAccessor */
-    protected $accessor;
+    protected PropertyAccessor $accessor;
+    protected IdentifiableObjectRepositoryInterface $categoryRepository;
+    private IsCategoryTreeLinkedToUser $isCategoryTreeLinkedToUser;
+    private IsCategoryTreeLinkedToChannel $isCategoryTreeLinkedToChannel;
 
-    /** @var IdentifiableObjectRepositoryInterface */
-    protected $categoryRepository;
-
-    /**
-     * @param IdentifiableObjectRepositoryInterface $categoryRepository
-     */
-    public function __construct(IdentifiableObjectRepositoryInterface $categoryRepository)
-    {
+    public function __construct(
+        IdentifiableObjectRepositoryInterface $categoryRepository,
+        IsCategoryTreeLinkedToUser $isCategoryTreeLinkedToUser,
+        IsCategoryTreeLinkedToChannel $isCategoryTreeLinkedToChannel
```

Comment:
```suggestion
        IsCategoryTreeLinkedToUser $isCategoryTreeLinkedToUser = null,
        IsCategoryTreeLinkedToChannel $isCategoryTreeLinkedToChannel = null
```

Otherwise it's a BC break for clients that have custom code.  
And you have to add a note to remove the null(s) on pull-up.

Reply:
Since its a backport from master, you would still put a note for it ?

Reply:
Maybe not necessary in this case